src/stream/src/executor/backfill.rs

```diff
@@ -188,102 +264,133 @@ where
         // finished.
         //
         // Once the backfill loop ends, we forward the upstream directly to the downstream.
-        'backfill_loop: loop {
-            let mut upstream_chunk_buffer: Vec<StreamChunk> = vec![];
-
-            let left_upstream = upstream.by_ref().map(Either::Left);
-
-            let right_snapshot = pin!(Self::snapshot_read(
-                &self.table,
-                snapshot_read_epoch,
-                current_pos.clone(),
-                true
-            )
-            .map(Either::Right),);
-
-            // Prefer to select upstream, so we can stop snapshot stream as soon as the barrier
-            // comes.
-            let backfill_stream =
-                select_with_strategy(left_upstream, right_snapshot, |_: &mut ()| {
-                    stream::PollNext::Left
-                });
-
-            let mut cur_barrier_snapshot_processed_rows: u64 = 0;
-            let mut cur_barrier_upstream_processed_rows: u64 = 0;
+        if to_backfill {
```

Comment:
Could you please remove this check? Because it must be true and the `backfill_loop` is too deep currently.